waffle_utils/video_processing/video_tools.py

```diff
@@ -0,0 +1,177 @@
+        self.input_path = self._input_path = input_path
+        self.output_path = self._output_path = output_path
+
+    # properties
+    @property
+    def input_path(self):
+        return self._input_path
+
+    @input_path.setter
+    @type_validator(str)
+    def input_path(self, v):
+        self._input_path = v
+
+    @property
+    def output_path(self):
+        return self._output_path
+
+    @output_path.setter
+    @type_validator(str)
+    def output_path(self, v):
+        self._output_path = v
+
+    # factories
+    @classmethod
+    def from_video(
+        cls,
+        input_path: str,
+        output_path: str,
+    ) -> "VideoTools":
+        """Create a `VideoTools` Instance from a Video File.
+
+        Args:
+            input_path (str): Path to the input video file.
+            output_path (str): Path to the output video or image files.
+
+        Raises:
+            ValueError: If the input file type is not supported.
+
+        Example:
+            # Create a VideoTools instance from a video file
+            vt = VideoTools.from_video(
+                "path/to/input.mp4",
+                "path/to/images",
+            )
+
+        Returns:
+            VideoTools: A VideoTools instance with the input and output paths.
+        """
+        # Check if input file is a supported video format
+        supported_formats = (".mp4", ".avi", ".mov", ".wmv")
+        if not input_path.lower().endswith(supported_formats):
+            raise ValueError(
+                f"Unsupported file type: {input_path}\n"
+                f"Supported file types are: {supported_formats}"
+            )
+
+        return cls(input_path, output_path)
+
+    @classmethod
+    def from_images(
+        cls,
+        input_path: str,
+        output_path: str,
+    ) -> "VideoTools":
+        """Create a `VideoTools` Instance from a Directory of images.
+
+        Args:
+            input_path (str): Path to the directory of input images.
+            output_path (str): Path to the output video file.
+
+        Raises:
+            ValueError: If the input file type is not supported.
+
+        Example:
+            # Create a VideoTools instance from a directory of images
+            vt = VideoTools.from_images(
+                "path/to/images",
+                "path/to/output.avi",
+            )
+
+        Returns:
+            VideoTools: A VideoTools instance with the input and output paths.
+        """
+        return cls(input_path, output_path)
+
+    def extract_frames(
+        self,
+        frame_rate: int = 30,
+        verbose: bool = False,
+    ) -> None:
+        """Extract Frames fom a Video
+
+        Args:
+            frame_rate (int, optional): Frame rate. Defaults to 30.
+            verbose (bool, optional): Verbose. Defaults to False.
+        """
+        # Create output directory if it doesn't exist
+        if not Path(self.output_path).exists():
+            os.makedirs(str(self.output_path))
```

Comment:
use file.io make_directory

Reply:
and consider the `create_directory` option.
Watch https://github.com/snuailab/waffle_utils/blob/223da43d2729e1f9ed26b8de6b9d4b7e56496baa/waffle_utils/file/io.py#L90